### DISCUSSION TITLE
Does it Support socks5://IP_ADDRESS:PORT@USER:PASS"?

### DISCUSSION THREAD

**Discussion by STARKILLER-1** [seleniumbase/SeleniumBase#3118] [Q&A] [answered]
Does it Support socks5://IP_ADDRESS:PORT@USER:PASS"?

**Comment by mdmintz** [comment, marked as answer]
Chromium doesn't support SOCKS5 authenticated proxies (https://issues.chromium.org/issues/40829748).

For socks4 or socks5 proxies, use these formats for the proxy_string:

Unauthenticated:

* ``socks4://IP_ADDRESS:PORT``

* ``socks5://IP_ADDRESS:PORT``

Authenticated:

* ``socks4://USER:PASS@IP_ADDRESS:PORT``

----

If you're using Wire Mode (which DOES NOT support UC Mode), there's a possible workaround:

```python
from seleniumbase import Driver

def set_proxy():
    driver.proxy = {
        "http": "socks5://user:pass@ip:port",
        "https": "socks5://user:pass@ip:port",
        "no_proxy": "localhost,127.0.0.1",
    }

driver = Driver(wire=True)
try:
    set_proxy()
    driver.get("https://ipinfo.io/")
    driver.sleep(5)
finally:
    driver.quit()
```

That works because `selenium-wire` sets up the proxy outside the browser on a locally-running server.

(Again, Wire Mode does not support UC Mode. Use `wire=True` to activate it.)

**Reply by STARKILLER-1** [reply]
Thank you very much!
